File: regex-test/java/Tester.java
import java.util.regex.Pattern;
import java.util.regex.PatternSyntaxException;

public class Tester {
    public static void main(String[] args) {
        String regex = args[1];
        try {
            var pattern = Pattern.compile("(?U)" + regex);
            for (int i = 2; i < args.length; i++) {
                var matcher = pattern.matcher(args[i]);
                if (matcher.matches()) {
                    var region = matcher.regionStart() + ".." + matcher.regionEnd();
                    System.err.println("[matches in " + region + "] " + args[i]);
                } else {
                    System.err.println("[no match] " + args[i]);
                }
            }
        } catch (PatternSyntaxException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}

File: regex-test/dotnet/Tester.cs
using System;
using System.Text.RegularExpressions;

class Tester {
    public static void Main(string[] args) {
        try {
            Regex r = new Regex(args[1], RegexOptions.Compiled);
            for (int i = 2; i < args.Length; i++) {
                MatchCollection matches = r.Matches(args[i]);
                if (matches.Count > 0) {
                    Match match = matches[0];
                    string region = match.Index + ".." + (match.Index + match.Value.Length);
                    Console.Error.WriteLine("[matches in {0}] {1}", region, args[i]);
                } else {
                    Console.Error.WriteLine("[no match] {0}", args[i]);
                }
            }
        } catch (ArgumentException e) {
            Console.Error.WriteLine(e.Message);
            Environment.Exit(1);
        }
    }
}
